Upgrade react native to v0.71.0

drewvolz opened the pull request:
## Upgrade from RN 0.68.5 ~> 0.71.0

- [Upgrade helper diff for 68.5 ~> 71.0 ](https://react-native-community.github.io/upgrade-helper/?from=0.68.5&to=0.71.0)
- [Upstream release notes](https://github.com/facebook/react-native/blob/main/CHANGELOG.md#v071)
- [Release blog announcement](https://reactnative.dev/blog/2023/01/12/version-071)
- Closes #5949
- Closes #5959

### Testing this out:
1. `git checkout drew/rn-71`
2. `npm i`
3. `cd ios && pod install` (with a handful of `repo-update` follow-ups)
4. `npm run start --reset-cache`
5. iOS: open .xcworkspace and build (you may have to comment out `npm run bundle-data` in the "bundle data files"  build phase, maybe my npm is not setup correctly)
6. Android: sync gradle, then build
7. Connect to metro

### 👀 Noteworthy
- Logbox ignore has been removed (we will now see our previously hidden logs)
- I removed the storyboard loading screen logic for iOS in the app delegate (but not the storyboard)

### 🎈 What looks to be working
- [Flipper](https://fbflipper.com)
- Navigation
- General network requests
- Components rendering / interacting

### ⚠️ What looks broken
- Search?
  - [x] Directory seems to work (tested on Android)
  - [x] Filter search (courses, a-to-z) seem broken (tested on Android)
- [x] Popover v5.0 fixes #6728 

### 📝 Todo
- [x] Merge #6572 into this branch
- [x] Remove `@types/react-native`
- [x] Lots of other dependencies should be upgraded
- [ ] Check that `ReactDatabaseSupplier` (async storage) on android is still working after changes to `MainActivity`
- [ ] How do we add both debug and release `ReactNativeFlipper.java` class? (to prevent flipper from running on prod)

drewvolz committed:
add segmented control dependency from react-native-community

drewvolz committed:
Merge branch 'master' into drew/segmented-control-dep-change

drewvolz committed:
add e2e to ignored jest test paths

drewvolz committed:
use test ignore patterns in jest config

drewvolz committed:
upgrade react and react-native

* react@18.2.0
* react-native@0.71.0

drewvolz committed:
add .node-version

drewvolz committed:
add .xcode.env

drewvolz committed:
update gitignore for android keystore and cxx

drewvolz committed:
remove jni directories

drewvolz committed:
(rn 71): flipper, hermes, new architecture

drewvolz committed:
(rn 71): gradle updates

* build tools, compileSdk, targetSdk up to 33
* ndk version of 23.1.7779620
* build gradle of 7.3.1
* remove gradle download task

drewvolz committed:
(rn 71): replace react database supplier import to come from community

drewvolz committed:
(rn 71): podfile updates for flipper

drewvolz committed:
(rn 71): logbox ignore was removed in the upgrade, cannot filter

drewvolz committed:
(rn 71): package updates

* upgrade async-storage
* upgrade react-native-reanimated
* upgrade react-native-safe-area-context

drewvolz committed:
(rn 71): podfile.lock updates

drewvolz committed:
Merge branch 'drew/rn-71' into drew/segmented-control-dep-change

drewvolz committed:
Merge pull request #6572 from StoDevX/drew/segmented-control-dep-change

add segmented control dependency from react-native-community

drewvolz committed:
(rn-71): use default entryfile config

drewvolz committed:
(rn-71): remove @types/react-native

drewvolz committed:
replace clipboard @react-native-clipboard/clipboard import

drewvolz committed:
alphabetize netinfo/clipboard deps

drewvolz committed:
update datetimepicker dep to 6.7.1

drewvolz committed:
update react-native-popover-view to 5.0.0 in filter module

drewvolz committed:
bump clipboard to 1.11.1

drewvolz committed:
metro-react-native-babel-preset to 0.73.7

drewvolz committed:
bump gesture handler to 2.8.0

drewvolz committed:
bump react-native-screens to 3.18.2

drewvolz committed:
remaining peer and babel-preset changes to the lockfile

drewvolz committed:
add safearea insets to popover

drewvolz committed:
filter section popover tweaks

wrap FilterSection in a view and apply the container styles to fix
rendering quirks from the popover library; this keeps the popover
contained and behaving as it did before for its content size.

drewvolz committed:
ready our string table stripping for when we use xcode 14 to release

without bitcode, our binary gets inflated without being compiled down by
apple via llvm and we are left with a lot of  DYLD.String Tables

these String Tables are the unnecessary metadata. We can reduce these in
a few ways in our app and our dependencies.

* cocoapods: we can strip the installed product on all pods.
* binary: we can strip flags and perform deployment post-processing

drewvolz committed:
upgrade react-native-popover-view to 5.1.7

rye commented on .gitignore:
```diff
@@ -22,6 +22,7 @@ DerivedData
 *.hmap
 *.ipa
 *.xcuserstate
+ios/.xcode.env.local
```
Can we anchor both this and the following `ios/` entries to the root `/ios` directory?
```suggestion
/ios/.xcode.env.local
```

drewvolz replied:
Sure!

drewvolz replied:
Regarding `gradle-wrapper.jar`: was in the diff helper to upgrade, changes are unknown.

rye reviewed (commented):
One gitignore suggestion and then a question: do we need to commit the `gradle-wrapper.jar` file?

drewvolz committed:
ready our string table stripping for when we use xcode 14 to release

without bitcode, our binary gets inflated without being compiled down by
apple via llvm and we are left with a lot of  DYLD.String Tables

these String Tables are the unnecessary metadata. We can reduce these in
a few ways in our app and our dependencies.

cocoapods: we can strip the installed product on all pods.
binary: we can strip flags and perform deployment post-processing

drewvolz committed:
prettier changes

drewvolz committed:
sort popover props alphabetically

drewvolz committed:
Merge pull request #6728 from StoDevX/drew/popover-updates

update react-native-popover-view

drewvolz committed:
anchor ios/ entries to the root /ios directory

co-authored-by: Kristofer Rye <kristofer.rye@gmail.com>

drewvolz committed:
Merge branch 'master' into drew/rn-71

drewvolz committed:
update FileReader.readAsArrayBuffer patch for 0.71

drewvolz committed:
remove datetimepicker manual linking, fixed by hoisting

drewvolz committed:
podfile.lock updates

drewvolz committed:
add note about segmentedcontrol ts error

drewvolz committed:
add typing for segmentedcontrol onchange event

drewvolz committed:
remove logbox import

drewvolz committed:
prettier formatting

hawkrives commented on package.json:
```diff
@@ -50,11 +50,12 @@
+    "@react-native-community/segmented-control": "2.2.2",
     "@react-native-picker/picker": "2.4.8",
-    "@react-navigation/bottom-tabs": "6.5.3",
+    "@react-navigation/bottom-tabs": "6.4.0",
```
Be sure not to keep this downgraded!

drewvolz replied:
Oop! Good catch

hawkrives committed:
switch segmented-control deps

this is the one that Expo ships

hawkrives committed:
update segmented-control import

hawkrives committed:
update types/react and types/react-test-renderer

hawkrives committed:
add patch for react-theming library

hawkrives committed:
work on removing glamorous

hawkrives committed:
add type hints and continue removing glamorous

hawkrives committed:
continue removing glamorous

hawkrives committed:
remove SegmentedControl from ApiTestIos in favor of a <Switch/>

hawkrives committed:
remove dep on rn-segmented-control

drewvolz committed:
update podfile.lock to remove segmented control

drewvolz committed:
Merge remote-tracking branch 'origin/master' into drew/rn-71

drewvolz committed:
revert downgrade of react-navigation/bottom-tabs

drewvolz committed:
Merge branch 'master' into drew/rn-71

hawkrives committed:
Merge branch 'master' into drew/rn-71

* master: (23 commits)
  use standard "black" color
  switch codecov from the bash uploader to their action
  add new "dependency review" workflow
  add whitespace to "check" workflow jobs
  make the Prettier job's ID match its name
  upgrade linux runners to ubuntu-22.04
  centralize global versions for languages and Xcode
  auto-cancel in-progress jobs on subsequent commit to the same branch
  add distributionSha256Sum to gradle-wrapper.properties
  implement auto-cache-cleanup workflow
  Bump react-native-network-logger from 1.13.0 to 1.14.0
  Bump @typescript-eslint/eslint-plugin from 5.48.2 to 5.49.0
  Bump jest from 29.3.1 to 29.4.0
  Bump @tanstack/react-query and @tanstack/react-query-persist-client
  clear username and password fields upon logout
  remove glamorous usages
  remove glamorous-native dependency
  Bump @tanstack/react-query and @tanstack/react-query-persist-client
  Bump @tanstack/query-async-storage-persister from 4.22.0 to 4.22.4
  Bump @babel/runtime from 7.20.7 to 7.20.13
  ...

drewvolz committed:
Merge branch 'master' into drew/rn-71

drewvolz committed:
update gradle lockfile version for gradle-download-task

hawkrives commented on android/app/build.gradle:
```diff
@@ -1,82 +1,58 @@
+    // hermesFlags = ["-O", "-output-source-map"]
+}
 
 def skipBundling = System.getenv().containsKey("SKIP_BUNDLING")
```
Was project.ext.react supposed to merge into the react {} block?

hawkrives commented on android/app/src/main/AndroidManifest.xml:
```diff
@@ -1,7 +1,6 @@
 <manifest
     xmlns:android="http://schemas.android.com/apk/res/android"
     xmlns:tools="http://schemas.android.com/tools"
-    package="com.allaboutolaf"
```
No more [package]? Neat!

hawkrives replied:
Ah,  I see: it's deprecated!

```
package="io.sentry.react" found in source AndroidManifest.xml: /Users/rives/Projects/allaboutolaf/node_modules/@sentry/react-native/android/src/main/AndroidManifest.xml.
Setting the namespace via a source AndroidManifest.xml's package attribute is deprecated.
Please instead set the namespace (or testNamespace) in the module's build.gradle file, as described here: https://developer.android.com/studio/build/configure-app-module#set-namespace
This migration can be done automatically using the AGP Upgrade Assistant, please refer to https://developer.android.com/studio/build/agp-upgrade-assistant for more information.
```

hawkrives commented on android/build.gradle:
```diff
@@ -1,10 +1,8 @@
 buildscript {
     ext {
-        buildToolsVersion = "31.0.0"
+        buildToolsVersion = "33.0.0"
```
Yay!

drewvolz committed:
Merge branch 'master' into drew/rn-71

# Conflicts:
#	modules/badge/package.json
#	modules/event-list/package.json
#	modules/filter/filter-popover.tsx
#	modules/filter/package.json
#	modules/food-menu/package.json
#	modules/icon/package.json
#	modules/lists/package.json
#	modules/navigation-buttons/package.json
#	modules/navigation-tabs/package.json
#	package-lock.json

drewvolz committed:
mock NativeEventEmitter in jest setup

drewvolz committed:
run prettier on jest-setup

drewvolz committed:
combine patches into a single file to fix application

passing a glob of patch files means you try to apply patches 1..n to the first
patch (0)

hawkrives committed:
clean up how we define Theme-s

hawkrives committed:
move a line

hawkrives committed:
re-add flipper plugins

hawkrives committed:
remove maybe-unused options from build.gradle?

hawkrives committed:
change where generated android assets live in the manual bundle command

hawkrives committed:
update android jsbundle caching

hawkrives committed:
cache the generated android app

hawkrives committed:
save the generated iOS app as an artifact

hawkrives committed:
remove glob expansion from android generation

hawkrives committed:
bundle on android again

hawkrives committed:
add prod ReactNativeFlipper

hawkrives committed:
poke at build.gradle
